### PR TITLE
feat: RFC improver CLI Error reporting

### PR DESCRIPTION
**Pull request by alamb**
~Draft as it builds on https://github.com/influxdata/influxdb_iox/pull/5734 (last commit)~

# Rationale
Inspired by @carols10cents 's comment in https://github.com/influxdata/influxdb_iox/pull/5734#pullrequestreview-1119028922 I tried to make error reporting better

# Changes
Using the Snafu [`iter_chain`](https://docs.rs/snafu/0.7.1/snafu/trait.ErrorCompat.html#method.iter_chain) I prototyped what it would look like to print out errors in this way

# Example output:

```shell
influxdb_iox debug parquet-to-lp influxdb_iox/src/main.rs

Error in parquet_to_lp subcommand: Error converting: Error inferring IOx schema: Parquet error: Parquet error: Invalid Parquet file. Corrupt footer
  Caused by: Error converting: Error inferring IOx schema: Parquet error: Parquet error: Invalid Parquet file. Corrupt footer
  Caused by: Error inferring IOx schema: Parquet error: Parquet error: Invalid Parquet file. Corrupt footer
  Caused by: Parquet error: Parquet error: Invalid Parquet file. Corrupt footer
```

While I do think this helps see some of the error structure better, it is still quite repetitive.

# Question / Musing
I could remove the redundancy by remove the `{source}` in the Error::Display "Error inferring IOx schema: {source}", but if I did so it would remove that information from other places the error is reported (e.g. in logs).

I wonder if it would be possible to implement the "Caused by" formatting automatically for the implementations of `Error::Display` so that I could simply do `println!("Error: {}", e)` and get a source chain? This would also allow the error to be logged as well

So for example, my ideal output would be:

```text
Error in parquet_to_lp subcommand: 
  Caused by: Error converting
  Caused by: Error inferring IOx schema
  Caused by: Parquet error: 
  Caused by: Invalid Parquet file
  Caused by: Corrupt footer
```

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: influxdb_iox/src/main.rs [hunk 1]
```diff
@@ -194,6 +194,21 @@ enum Command {
...
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use snafu::ErrorCompat;
+
+        write!(f, "{}", self.0)?;
+        for cause in ErrorCompat::iter_chain(&self.0).skip(1) {
+            write!(f, "  Caused by: {cause}")?;
+        }
+        Ok(())
+    }
+}
```
Comment (shepmaster) [inline influxdb_iox/src/main.rs:217]: This is pretty much the same as [what I'm working on adding to SNAFU in the next release](https://github.com/shepmaster/snafu/commit/8730358de8f7bb5b506058485d05564f671d162f#diff-667bdcc3d21a418a8806ad64a3cb61f2ca3e778361aa79e07dfb70a709dd5a46R203-R228)

The only comment I have now is that you could probably avoid using `snafu::ErrorCompat` altogether here by using `ChainCompat` directly, something like `ChainCompat::new(&self.0).skip(1)`.

File: influxdb_iox/src/main.rs [hunk 2]
```diff
@@ -194,6 +194,21 @@ enum Command {
...
 }
 
+/// Prints the error in a heirarchy to stderr
+struct FormattedError<E>(E);
+
+impl<E: snafu::ErrorCompat + std::error::Error + 'static> Display for FormattedError<E> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use snafu::ErrorCompat;
+
+        write!(f, "{}", self.0)?;
```
Comment (shepmaster) [inline influxdb_iox/src/main.rs]: Why the choice of `write!` instead of `writeln!`?

Comment (shepmaster) [reply influxdb_iox/src/main.rs]: 🙈

Comment (alamb) [reply influxdb_iox/src/main.rs]: I think it is an oversight. I will fix

Comment (alamb) [reply influxdb_iox/src/main.rs]: Sorry I got too eager asking for feedback

**Comment by alamb** [comment]
Superceded by https://github.com/influxdata/influxdb_iox/pull/6050

**Pull request closed without merging**